components/prism-emacs.js

```diff
@@ -0,0 +1,179 @@
+Prism.languages.emacs = (function () {
+	// Patterns in regular expressions
+
+	// Symbol name. See https://www.gnu.org/software/emacs/manual/html_node/elisp/Symbol-Type.html
+	// & and : are excluded as they are usually used for special purposes
+	const symbol = '[-+*/_~!@$%^=<>{}\\w]+';
+	// symbol starting with & used in function arguments
+	const marker = '&' + symbol;
+	// Open parenthesis for look-behind
+	const par = '(\\()';
+	const endpar = '(?=\\))';
+	// End the pattern with look-ahead space
+	const space = '(?=\\s)';
+
+	// Functions to construct regular expressions
+	// simple form
+	// e.g. (interactive ... or (interactive)
+	const simple_form = name => new RegExp(`(\\()${name}(?=[\\s\\)])`);
+	// booleans and numbers
+	const primitive = pattern => new RegExp(`([\\s\\(\\[])${pattern}(?=[\\s\\)])`);
+
+	var language = {
+		// Three or four semicolons are considered a heading.
+		// See https://www.gnu.org/software/emacs/manual/html_node/elisp/Comment-Tips.html
+		'heading': {
+			pattern: /;;;.*/,
+			alias: ['comment', 'title']
+		},
+		'comment': /;.*/,
+		'string': {
+			pattern: /"(?:[^"\\]*|\\.)*"/,
+			greedy: true,
+			inside: {
+				'argument': /[-A-Z]+(?=[.,\s])/,
+				'symbol': new RegExp("`" + symbol + "'")
+			}
+		},
+		'quoted-symbol': {
+			pattern: new RegExp('#?\'' + symbol),
+			alias: ['variable', 'symbol']
+		},
+		'lisp-property': {
+			pattern: new RegExp(':' + symbol),
+			alias: 'property'
+		},
+		'splice': {
+			pattern: new RegExp(",@?" + symbol),
+			alias: ['symbol', 'variable']
+		},
+		'keyword': [
+			{
+				pattern: new RegExp(par + "(?:(?:lexical-)?let\\*?|(?:cl-)?letf|if|when|while|unless|cons|cl-loop|and|or|not|cond|setq|error|message|null|require|provide|use-package)" + space),
+				lookbehind: true
+			},
+			{
+				pattern: new RegExp(par + "(?:for|do|collect|return|finally|append|concat|in|by)" + space),
+				lookbehind: true
+			}
+		],
+		'declare': {
+			pattern: simple_form("declare"),
+			lookbehind: true,
+			alias: 'keyword'
+		},
+		'interactive': {
+			pattern: simple_form("interactive"),
+			lookbehind: true,
+			alias: 'keyword'
+		},
+		'boolean': {
+			pattern: primitive("(?:t|nil)"),
+			lookbehind: true
+		},
+		'number': [
+			{
+				pattern: primitive("[-+]?\\d+(?:\\.\\d*)?"),
+				lookbehind: true
+			}
+		],
+		'defvar': {
+			pattern: new RegExp(par + 'def(?:var|const|custom|group)\\s+' + symbol),
+			lookbehind: true,
+			inside: {
+				'keyword': /def[a-z]+/,
```

Comment:
I wonder what the highlighting would be here with a code piece like `(defvar defabc`? Maybe a `^` anchor should be added at the beginning of this regexp.